asmcomp/cmmgen.ml

```diff
@@ -2030,6 +2052,33 @@ let rec transl env e =
       let dbg = Debuginfo.none in
       Cop(Cload (Word_int, Mutable), [Cconst_int 0], dbg)
 
+and transl_catch env nfail ids body handler =
+  let env_handler, typs, ids_with_types =
+    List.fold_right
+      (fun (id, k) (env, typs, ids) ->
+         match k with
+         | Pfloatval ->
+             let dbg = Debuginfo.none in
+             let unboxed_id = V.create_local (VP.name id) in
+             let typ = Boxed_float dbg in
+             add_unboxed_id (VP.var id) unboxed_id typ env,
+             Some typ :: typs,
+             (* keep provenance from id? *)
+             (VP.create unboxed_id, Cmm.typ_float) :: ids
```

Comment:
That unboxing might be too aggressive. it doesn't take the use into account for the decision.